tensorflow/core/kernels/segment_reduction_ops_impl.h

```diff
@@ -957,6 +957,125 @@ class SparseSegmentSqrtNGradOp
                                                      true /*is_sqrtn*/) {}
 };
 
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+template <class Device, class T, class Tindex, bool has_num_segments>
+class SparseSegmentSumGpuOp : public AsyncOpKernel {
+ public:
+  explicit SparseSegmentSumGpuOp(OpKernelConstruction* context)
+      : AsyncOpKernel(context){};
+
+  void ComputeAsync(OpKernelContext* context, DoneCallback done) override {
+    const Tensor& input_data = context->input(0);
+    const Tensor& indices = context->input(1);
+    const Tensor& segment_ids = context->input(2);
+
+    OP_REQUIRES_ASYNC(context, TensorShapeUtils::IsVector(indices.shape()),
+                      errors::InvalidArgument("indices should be a vector."),
+                      done);
+    const int64 num_indices = indices.NumElements();
+
+    OP_REQUIRES_ASYNC(
+        context, TensorShapeUtils::IsVector(segment_ids.shape()),
+        errors::InvalidArgument("segment_ids should be a vector."), done);
+
+    OP_REQUIRES_ASYNC(
+        context, num_indices == segment_ids.NumElements(),
+        errors::InvalidArgument("indices and segment_ids should have"
+                                "same length."),
+        done);
+
+    ScratchSpace<Index> output_rows_host(context, 1, /* on_host */ true);
```

Comment:
Please use the following spelling:

```
ScratchSpace<Index> output_rows_host(context, /*size=*/1, /*on_host=*/ true);
```

Our internal tooling recognizes this format and will check that the parameter names actually match up.

Reply:
[Accepted](https://github.com/tensorflow/tensorflow/pull/45085/commits/a119214973391761cda0c74698ed239456361b98#diff-d29812caca45335066ddacf8fb06afc416f1f557bc446068ebaa163495e0b03bR987)